src/pm/xbps.rs

```diff
@@ -48,14 +49,64 @@ impl Pm for Xbps {
 
     /// Q generates a list of installed packages.
     async fn q(&self, kws: &[&str], flags: &[&str]) -> Result<()> {
-        self.run(Cmd::new(&["xbps-query", "-l"]).kws(kws).flags(flags))
-            .await
+        if kws.is_empty() {
+            return self
+                .run(Cmd::new(&["xbps-query", "-l"]).kws(kws).flags(flags))
+                .await;
+        }
+
+        let results = stream::iter(kws)
+            .filter_map(|&pkg| async {
+                let cmd = Cmd::new(&["xbps-query", "--property", "pkgver", pkg]).flags(flags);
+                self.check_output(cmd, PmMode::Mute, &Strategy::default())
+                    .await
+                    .map(|res| String::from_utf8(res).unwrap())
```

Comment:
Nicely done. There are still some places to be fixed though:

- There's no need to convert to `String` though: `tokio::io::Stdout` can receive `&[u8]` directly. 
- Also, by using `.ok()`, you're actually losing some error messages that should have been passed.

I thus propose:

```rust
use tokio::io;

async fn q(&self, kws: &[&str], flags: &[&str]) -> Result<()> {
    use tokio::io::AsyncWriteExt;

    if kws.is_empty() {
        return self
            .run(Cmd::new(&["xbps-query", "-l"]).kws(kws).flags(flags))
            .await;
    }

    let lines: Vec<Vec<u8>> = stream::iter(kws)
        .map(Ok)
        .and_then(|&pkg| async {
            let cmd = Cmd::new(&["xbps-query", "--property", "pkgver", pkg]).flags(flags);
            self.check_output(cmd, PmMode::Mute, &Strategy::default())
                .await
        })
        .try_collect()
        .await?;

    stream::iter(lines)
        .map(Ok)
        .try_fold(io::stdout(), |mut out, ln| async move {
            out.write_all(&ln).await?;
            Ok(out)
        })
        .await
        .map(|_| ())
        .map_err(Error::IoError)
}
```

Reply:
This discards all the query results if any of the packages is not found, while `pacman -Q` lists each error/success independently. I'm wondering if there's a way to mimic that.

Reply:
@tranzystorek-io 
Sorry for answering you this late (it seems that you didn't mention me so I didn't receive any notification).

In that case, it's completely possible to match against from a `Error::CmdStatusCodeError { code, output }` in `Result<Output>` that you've got, recover the error if necessary, collecting a collection of output until the error type becomes something else.

Reply:
Finally found some time to tweak that code 😅

The `q()` and `qe()` are similar but I'm not sure if they can be refactored easily...